Assets/SEE/VCS/GitVersionControl.cs

```diff
@@ -0,0 +1,187 @@
+using LibGit2Sharp;
+using System.Linq;
+using System.IO;
+using System;
+using System.Collections.Generic;
+using UnityEngine.Assertions;
+using UnityEngine;
+
+namespace SEE.VCS
+{
+    /// <summary>
+    /// Implements the functionality of <see cref="IVersionControl"/> for Git.
+    /// </summary>
+    internal class GitVersionControl : IVersionControl
+    {
+        /// <summary>
+        /// Constructor setting up the repository access.
+        /// </summary>
+        /// <param name="repositoryPath">the path to a Git repository</param>
+        /// <exception cref="ArgumentNullException">throw in case <paramref name="repositoryPath"/>
+        /// is null or just whitespace</exception>
+        internal GitVersionControl(string repositoryPath)
+        {
+            if (string.IsNullOrWhiteSpace(repositoryPath))
+            {
+                throw new ArgumentNullException("Repository path must not be null or empty.");
+            }
+            this.repositoryPath = repositoryPath;
+            repo = new(repositoryPath);
+        }
+
+        /// <summary>
+        /// The repository.
+        /// </summary>
+        private readonly Repository repo;
+
+        /// <summary>
+        /// The path to the root of the repository.
+        /// </summary>
+        private readonly string repositoryPath;
+
+        /// <summary>
+        /// See <see cref="IVersionControl.Show(string, string)"/>.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">throw in case <paramref name="fileName"/>
+        /// or <paramref name="commitID"/> are null or just whitespace</exception>
+        public string Show(string fileName, string commitID)
+        {
+            if (string.IsNullOrWhiteSpace(fileName))
+            {
+                throw new ArgumentNullException("Path must not be null or empty.");
```

Comment:
`ArgumentNullException` takes as its parameter the name of the parameter that is null and then constructs a message around it. For example, here it should be `ArgumentNullException(nameof(fileName))`. This applies to other places in the PR too.

Reply:
It should have been `ArgumentException`. Apparently, I didn't pay attention to the code completion.

Reply:
I fixed it.